.github/workflows/pull_request.yaml

```diff
@@ -7,31 +7,31 @@ on:
 
 env:
   DOCKER_FILE_PATH: Dockerfile
-  GOLANG_VERSION: 1.20.3
   KUBERNETES_VERSION: "1.18.0"
   KIND_VERSION: "0.10.0"
   REGISTRY: ghcr.io
 
 jobs:
   build:
-    runs-on: ubuntu-latest 
+    runs-on: ubuntu-latest
     name: Build
-    if: "! contains(toJSON(github.event.commits.*.message), '[skip-ci]')"    
+    if: "! contains(toJSON(github.event.commits.*.message), '[skip-ci]')"
     steps:
     - name: Check out code
       uses: actions/checkout@v3
       with:
         ref: ${{github.event.pull_request.head.sha}}
-        
+
     # Setting up helm binary
     - name: Set up Helm
       uses: azure/setup-helm@v3
 
     - name: Set up Go
-      id: go
-      uses: actions/setup-go@v3
+      uses: actions/setup-go@v4
       with:
-        go-version: ${{ env.GOLANG_VERSION }}
+        go-version-file: 'go.mod'
+        check-latest: true
```

Comment:
@Moulick do you know if specifying `check-latest` will conflict with specifying a go-version-file? Let's say go.mod specifies 1.19, and latest go version is 1.20, how would that work?

Reply:
As far as I understand and have observed in my own repos, it should stick to geting the latest patch version within the specified minor version. I'll do a quick check in a separate repo just to double check the behaviour
https://github.com/actions/setup-go#check-latest-version

Reply:
@karl-johan-grahn I checked just now. The `check-latest` happens exactly as I wrote above.

https://github.com/Moulick/ingress-whitelister/compare/renovate/github.com-cloudflare-cloudflare-go-0.x...latest-check
| step | go.mod | version downloaded | link | 
|--------|--------|--------| -------| 
| 1 | 1.20 | 1.20.4 | https://github.com/Moulick/ingress-whitelister/actions/runs/4934885937/jobs/8820484394#step:3:11 |
| 2 | 1.19 | 1.19.9 | https://github.com/Moulick/ingress-whitelister/actions/runs/4934761814/jobs/8820202900#step:3:11 |
| 3 | 1.20 | 1.20.4 |  https://github.com/Moulick/ingress-whitelister/actions/runs/4934816243/jobs/8820322356#step:3:11 |
| 4 | 1.19 | 1.19.9 | https://github.com/Moulick/ingress-whitelister/actions/runs/4934885937/jobs/8820484394#step:3:11 |